lifecycle.md

```diff
@@ -48,9 +48,44 @@ Note: this content should be moved to another location as it is not lifecycle re
 
 Build and test images are referenced in repo infra files, across a variety of `main` and `release/*` branches. Updating these references is a multi-step detail-oriented task. It is a pain, but necessary.
 
-We use version specific references in infra to ensure that our CI builds are reliable. One can imagine using floating OS tags (such as `debian-oldest` and `debian-latest`), however such an approach would be guaranteed to break our build. We know that since we often see build and test breaks that need addressing in PRs where we update build and test images.
+Two types of tag styles are available: [distro version-specific](#distro-version-specific-tags) and [floating](#floating-tags).
 
-At times, it may be necessary to use a [fixed image reference](https://github.com/dotnet/runtime/pull/110199#discussion_r1859075989) for build reliability. If this is ever done, a tracking issue should be created (before the PR is merged) so that we remember to resolve the underlying issue and update the image reference.
+### Distro Version-Specific Tags
+
+Distro version-specific tags include the distro's version (e.g. `alpine-3.21-helix-amd64`).
+Repos that are susceptible to breaking changes in the distro should use these tags.
+[dotnet/runtime](https://github.com/dotnet/runtime) is an example of such a repo.
+
+### Floating Tags
+
+Floating tags have no distro version indicated in the name and are scoped to a .NET version (e.g. `alpine-net9.0-helix-amd64`).
+It is routinely updated to reference a new version as the distro's and .NET's lifecycles progress.
+
+Floating tags are provided on an as-needed basis.
+They are beneficial for repos that are not susceptible to breaking changes that occur from new distro versions because the source that references the tag doesn't need to be updated in order to make use of the new version.
+
+#### Moving the Floating Tag
+
+There is a workflow that is following before a floating tag gets moved to a newer distro version:
+1. First, version-specific tags for the new distro version are provided.
+1. After a one month evaluation period, the floating tag is ready to reference the new version, assuming there are no issues found.
```

Comment:
Would it be worthwhile to update the in-development version ahead of servicing versions of .NET?  e.g. updating the .NET 10 alpine image before updating the .NET 9.0 + 8.0 images.  This could help provide stability in the servicing versions.

Reply:
That is an interesting idea. That likely makes sense.

Reply:
Can you write up a proposed timeline? I'm not following why we need two months but I might be missing something.

Reply:
It was based on Michael's suggestion:

> updating the .NET 10 alpine image before updating the .NET 9.0 + 8.0 images.

If not 2 months, then what is the timeframe between updating .NET 10 and the others?

Reply:
Here is what I interpreted by the suggestion, for a timeline:

- July: Alpine 3.22 ships
- August:
  - .NET 10 Preview -- Switches to Alpine 3.22 as only option (for both version-specific and floating tags)
  - .NET 8/9 -- Both 3.21 and 3.22 are available as version-specific options; floating stays on 3.21
- September: 
  - .NET 8/9 -- Both 3.21 and 3.22 are available as version-specific options; floating stays is updated to 3.22 

Perhaps @MichaelSimons meant something different.

Reply:
I was originally thinking a shorter timeline (2 week in main).  I don't see a particular driving reason for that.  The two month rollout it amenable to me.

Reply:
@richlander - Where does the runtime evaluation period fit into your timeline? Are you suggesting that floating tags for .NET 10 Preview get updated prior to a 1 month evaluation period in runtime? Or does that evaluation period start in July?

Reply:
Good question. I forgot about that. Let's assume that July (in this example) is "runtime signoff".